src/Debugger/DebuggerClient.php

```diff
@@ -0,0 +1,135 @@
+
+/**
+ * Google Stackdriver Debugger allows you to collect variable data from a live application
+ * and display it in the Google Cloud Platform Console. Find more information at
+ * [Stackdriver Debugger API docs](https://cloud.google.com/debugger/docs/).
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Debugger\DebuggerClient;
+ *
+ * $debugger = new DebuggerClient();
+ * ```
+ */
+class DebuggerClient
+{
+    use ClientTrait;
+
+    const VERSION = '0.1.0';
+    const DEFAULT_AGENT_VERSION = 'google.com/gcp-php/v0.1';
+
+    const FULL_CONTROL_SCOPE = 'https://www.googleapis.com/auth/cloud-platform';
+    const READ_ONLY_SCOPE = 'https://www.googleapis.com/auth/debugger.readonly';
+
+    /**
+     * @var ConnectionInterface $connection Represents a connection to Debugger
+     */
+    protected $connection;
+
+    /**
+     * Create a Debugger client.
+     *
+     * @param array $config [optional] {
+     *     Configuration options.
+     *
+     *     @type string $projectId The project ID from the Google Developer's
+     *           Console.
+     *     @type CacheItemPoolInterface $authCache A cache used storing access
+     *           tokens. **Defaults to** a simple in memory implementation.
+     *     @type array $authCacheOptions Cache configuration options.
+     *     @type callable $authHttpHandler A handler used to deliver Psr7
+     *           requests specifically for authentication.
+     *     @type callable $httpHandler A handler used to deliver Psr7 requests.
+     *           Only valid for requests sent over REST.
+     *     @type array $keyFile The contents of the service account credentials
+     *           .json file retrieved from the Google Developer's Console.
+     *           Ex: `json_decode(file_get_contents($path), true)`.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type int $retries Number of retries for a failed request.
+     *           **Defaults to** `3`.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::FULL_CONTROL_SCOPE];
+        }
+
+        $this->connection = new Rest($this->configureAuthentication($config));
+    }
+
+    /**
+     * Lazily instantiate a debuggee. There are no network requests made at this
+     * point. To see the operations that can be performed on a debuggee, please
+     * see {@see Google\Cloud\Debugger\Debuggee}.
+     *
+     * Example:
+     * ```
+     * $debuggee = $client->debuggee();
+     * $debuggee->register();
+     * ```
+     *
+     * ```
+     * $debuggee = $client->debuggee('debuggee-id');
+     * ```
+     *
+     * @param string $id The debuggee identifier
+     * @param array $extras [optional]
```

Comment:
What are extras intended to hold? The convention we use everywhere else is `$options`.

Reply:
It's the extra configuration options for the debuggee. I'll switch to `$options` and document them.